### ISSUE TITLE
APNS does not return a response

### ISSUE THREAD

**Issue by mrinterweb** [NicosKaralis/pushmeup#73]
I am trying to write code that tracks the delivery of push notifications. The response I always get from calling `APNS.send_notification` is always `nil`. `GCM.send_notification` gives me a response hash. With just a nil response, I have no way of knowing if the notification was delivered for APNS. In development the push notifications are delivered, still with a `nil` response. In production (with TestFlight, the production gateway URL, and my production pem), notifications are not delivered and the response is still `nil`. I don't know what to do if I can't verify the response from `APNS.send_notification`. 

Here is the code where I am calling `APNS.send_notification`:

``` ruby
    def push_to_ios_clients
      was_successful = true
      APNS.start_persistence
      pending_ios_clients.each do |client|
        result = APNS.send_notification(client.push_id, apns_data)
        logger.info "APNS response: #{result.inspect}" # result is always nil
        # todo - need to be able to verify response to determine if notification was sent
      end
      was_successful
    ensure
      APNS.stop_persistence
    end
```

**Comment by mklb** [comment]
I have the same problem. If I find a solution I´ll post it

**Comment by mklb** [comment]
so in general - APNS does not response anything. It´s a black box. So nil will always be the response.

My problem was that I didn´t have a matching provisioning profile and certificates. So my advice would be to check xcode and the apple developer portal out.

(a good tutorial for the certificate stuff: http://quickblox.com/developers/How_to_create_APNS_certificates - Always helps me to remember these horrible steps)

Hope I could help ;D

**Comment by mrinterweb** [comment]
Thanks @mklb. After doing more research into how the APNS protocol works, I began to realize how getting a response from the binary protocol would not be possible (or at least I could not find any APNS docs on getting a response). It is just fire-and-forget. I dislike services like this, and I hope that one day Apple improves the APNS API. Apple should take inspiration from Google's GCM.

I eventually realized that the problem I was having with delivering push notifications in production was that I wasn't using the correct provisioning profile. There are so many steps in the complicated dance that is APNS configuration.

I'm going to close this issue, as I don't think there can be a meaningful resolution. PushMeUp could return `true` instead of `nil`, but really it would not make much of a difference as there is no response from APNS in the first place.
